src/Lib.P2P.Tests/Transports/TcpTest.cs

```diff
@@ -0,0 +1,232 @@
+                Assert.AreEqual(listenerAddress, local);
+                Assert.IsNotNull(remote);
+                Assert.AreNotEqual(local, remote);
+                connected = true;
+            }
+
+            try
+            {
+                listenerAddress = tcp.Listen("/ip4/127.0.0.1", Handler, cs.Token);
+                Assert.IsTrue(listenerAddress.Protocols.Any(p => p.Name == "tcp"));
+                await using (var stream = await tcp.ConnectAsync(listenerAddress, cs.Token))
+                {
+                    await Task.Delay(50, cs.Token);
+                    Assert.IsNotNull(stream);
+                    Assert.IsTrue(connected);
+                }
+            }
+            finally
+            {
+                cs.Cancel();
+            }
+        }
+
+        [TestMethod]
+        public async Task Listen_Handler_Throws()
+        {
+            var tcp = new Tcp();
+            var cs = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+            var called = false;
+
+            void Handler(Stream stream, MultiAddress local, MultiAddress remote)
+            {
+                called = true;
+                throw new Exception("foobar");
+            }
+
+            try
+            {
+                var addr = tcp.Listen("/ip4/127.0.0.1", Handler, cs.Token);
+                Assert.IsTrue(addr.Protocols.Any(p => p.Name == "tcp"));
+                await using (var stream = await tcp.ConnectAsync(addr, cs.Token))
+                {
+                    await Task.Delay(50, cs.Token);
+                    Assert.IsNotNull(stream);
+                    Assert.IsTrue(called);
+                }
+            }
+            finally
+            {
+                cs.Cancel();
+            }
+        }
+
+        [TestMethod]
+        public async Task SendReceive()
+        {
+            var cs = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+            var tcp = new Tcp();
+            
+            using (var server = new HelloServer())
+            {
+                await using (var stream = await tcp.ConnectAsync(server.Address, cs.Token))
+                {
+                    var bytes = new byte[5];
+                    await stream.ReadAsync(bytes, 0, bytes.Length, cs.Token);
+                    Assert.AreEqual("hello", Encoding.UTF8.GetString(bytes));
+                }
+            }
+        }
+
+        private sealed class HelloServer : IDisposable
+        {
+            private readonly CancellationTokenSource _cs = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+
+            public HelloServer()
+            {
+                var tcp = new Tcp();
+                Address = tcp.Listen("/ip4/127.0.0.1", Handler, _cs.Token);
+            }
+
+            public MultiAddress Address { get; }
+
+            public void Dispose() { _cs.Cancel(); }
+
+            private static void Handler(Stream stream, MultiAddress local, MultiAddress remote)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'remote'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)